specs/Deployment/DeploymentAPI.md

```diff
@@ -179,12 +186,43 @@ they shut down, to refer to the updated framework package.
     }
 ```
 
+## Repair
+
+This API performs a status check, and if the main and singleton packages are already installed at
+the required version, it will attempt to repair those packages. All information about the version,
+channel, and architecture needed are derived from the current WinAppSDK framework package.
+
+Since both the main _and_ the singleton packages may need to be repaired, it is possible that one
+package is repaired successfully but not the other. If so, the returned WindowsAppRuntimeStatus will
+contain the error, if one occurs during package install.
+
+```C#
+    if (DeploymentManager.GetStatus().Status != DeploymentStatus.Ok)
+    {
+        // Repair does a status check, and if the status is OK it will attempt to get
+        // the WindowsAppRuntime into a good state by deploying packages. Unlike a simple
+        // status check, Repair can sometimes take several seconds to deploy the packages.
+        // These should be run on a separate thread so as not to hang your app while the
+        // packages deploy.
```

Comment:
Is this any different than Initialize()?

Reply:
No difference, here.

Only difference is in functionality that Initialize installs both packages even if either of them is not installed. Repair installs both packages only when both packages are already installed. So, their functionalities are mutually exclusive.

Reply:
So when should I use --repair? Only when I know WinAppRuntime is installed but maybe broken?

If e.g. Main is registered for the user "windowsappruntimeinstall.exe --repair" won't do anything with it, even if Main's package status != OK?


Sounds kinda backwards. `Install` today is installing what's missing AND re-registering what's already present (which is a form of repair) whereas `Repair` only installs what's not installed. And both consider "installed" to mean "package is registered for user" regardless if it's actually working.

Further, "broken" has 2 degrees of "broken"

1. Known and tagged as 'broken' (PackageStatus != OK)
2. Not explicitly known to Windows as 'broken' but no less busted and needing repair (e.g. register package then delete some regkeys laid down by the install; package status can be OK but the install is busted and in need of repair)

Seems we need more permutations:

1. Install only what's not installed
2. Install what's not installed AND fix packages with package status != ok
3. Install everything (whether or not it's already installed)

Today you're saying 1=`--repair`, 3=`--install` (i.e. default / no --repair) and 2=doesn't exist

Thoughts?

Reply:
Per our offline discussion on this topic,
repair = Install everything (--repair option for installer)
initialize = Install everything, if even one of the WindowsAppSDK packages is not installed or is not in good state.